src/main/webapp/app/course/competencies/competency-accordion/competency-accordion.component.ts

```diff
@@ -89,23 +90,36 @@ export class CompetencyAccordionComponent implements OnChanges {
             this.nextExercises = [];
         }
 
-        const submittedExercises = Object.keys(this.metrics.exerciseMetrics?.latestSubmission ?? {}).map(Number);
-        const competencyExercises = this.metrics.competencyMetrics?.exercises?.[this.competency.id] ?? [];
-        const nextExerciseInformations = competencyExercises
-            .filter((exerciseId) => !submittedExercises.includes(exerciseId))
-            .flatMap((exerciseId) => this.metrics.exerciseMetrics?.exerciseInformation?.[exerciseId] ?? [])
-            .filter((exercise) => exercise.startDate.isBefore(dayjs()) && (!exercise.dueDate || exercise.dueDate.isAfter(dayjs())))
-            .sort((a, b) => (a.dueDate ?? a.startDate).diff(b.dueDate))
-            .slice(0, 5);
-
-        // Workaround to convert ExerciseInformation to Exercise
-        this.nextExercises = nextExerciseInformations.map(
-            (exercise) =>
-                ({
-                    ...exercise,
-                    studentAssignedTeamIdComputed: exercise.studentAssignedTeamId !== undefined,
-                }) as unknown as Exercise,
+        const courseExercises = this.course?.exercises ?? [];
+        const exerciseIdToExercise = Object.fromEntries(courseExercises.map((exercise) => [exercise.id, exercise] as [number, Exercise]));
+        const activeCompetencyExercises = (this.metrics.competencyMetrics?.exercises?.[this.competency.id] ?? [])
+            .flatMap((exerciseId) => [exerciseIdToExercise[exerciseId]] ?? [])
+            .filter((exercise) => exercise.releaseDate?.isBefore(dayjs()))
```

Comment:
```suggestion
            .filter((exercise) => !exercise.dueDate || exercise.releaseDate.isBefore(dayjs()))
```
Currently all exercises without a release date are filtered out

Reply:
I do not think this would solve this problem. An exercise can have a due date without having a release date, so it would still filter out those exercises.

Reply:
You're right, it was supposed to be the release Date each time:
```suggestion
            .filter((exercise) => ! exercise.releaseDate || exercise.releaseDate?.isBefore(dayjs()))
```